### PR TITLE
applications: hide "Get ____" launchers

### PR DESCRIPTION
**Pull request by wjt**
These are not "real" applications, so we should hide them from the
Applications panel (just as we do for web apps).

https://phabricator.endlessm.com/T27105

### CODE REVIEW

File: panels/applications/cc-applications-panel.c [hunk 1]
```diff
@@ -1425,7 +1438,8 @@ populate_applications (CcApplicationsPanel *self)
       id = get_app_id (info);
       if (!g_app_info_should_show (info) ||
           !mct_app_filter_is_appinfo_allowed (self->app_filter, info) ||
-          g_str_has_prefix (id, EOS_LINK_PREFIX))
+          g_str_has_prefix (id, EOS_LINK_PREFIX) ||
+          get_replaced_by (info) != NULL)
```
Comment (GeorgesStavracas) [inline panels/applications/cc-applications-panel.c]: This is leaking the returned string.

File: panels/applications/cc-applications-panel.c [hunk 2]
```diff
@@ -1405,6 +1405,19 @@ update_panel (CcApplicationsPanel *self,
...
+ * app is not installed, or forward to the app if it is. Fetch the key which
+ * indicates which app replaces @info, if any, or %NULL otherwise.
+ */
+static char *
+get_replaced_by (GAppInfo *info)
+{
+  if (G_IS_DESKTOP_APP_INFO (info))
+    return g_desktop_app_info_get_string (G_DESKTOP_APP_INFO (info), "X-Endless-Replaced-By");
+
+  return NULL;
```
Comment (GeorgesStavracas) [inline panels/applications/cc-applications-panel.c]: Do you plan to use this function somewhere else? If not, I think returning a boolean and inverting the checked condition makes more sense. Something like:

```c
static gboolean
get_replaced_by (GAppInfo *info)
{
  g_autofree gchar *replaced_by = NULL;

  if (!G_IS_DESKTOP_APP_INFO (info))
    return FALSE;

  replaced_by = g_desktop_app_info_get_string (G_DESKTOP_APP_INFO (info), "X-Endless-Replaced-By");
  return replaced_by != NULL;
}
```

File: panels/applications/cc-applications-panel.c (hunk 1 above)
Comment (wjt) [reply panels/applications/cc-applications-panel.c]: Good catch.

File: panels/applications/cc-applications-panel.c (hunk 2 above)
Comment (wjt) [reply panels/applications/cc-applications-panel.c]: Yep, that's a good idea. I used a different name and slightly different (but equivalent) control flow.

**Pull request merged**